website_lazy_load_image/static/src/js/frontend.js

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2018 Onestein
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+
+odoo.define('website_lazy_load_image.lazy_image_loader', function (require) {
+    'use strict';
+
+    var Class = require('web.Class');
+    var mixins = require('web.mixins');
+
+    var LazyImageLoader = Class.extend(mixins.EventDispatcherMixin, {
+        /**
+         * @constructor
+         */
+        init: function () {
+            mixins.EventDispatcherMixin.init.call(this);
+            this.all_finished = $.Deferred();
+            var selector = '#wrapwrap > main img, #wrapwrap > footer img';
+            this.plugin = $(selector).data('loaded', false).lazy(
+                this._getPluginConfiguration()
+            );
+        },
+
+        /**
+         * Get the settings for the initialization of the lazy loading plugin.
+         *
+         * @private
+         * @returns {Object} Lazy loading plugin settings
+         */
+        _getPluginConfiguration: function () {
+            return {
+                afterLoad: this._afterLoad.bind(this),
+                beforeLoad: this._beforeLoad.bind(this),
+                onError: this._onError.bind(this),
+                onFinishedAll: this._onFinishedAll.bind(this),
+                chainable: false,
+            };
+        },
+
+        /**
+         * Triggered by the beforeLoad event of the lazy loading plugin.
+         *
+         * @param {DOMElement} el
+         * @private
+         */
+        _beforeLoad: function (el) {
+            this.trigger('beforeLoad', el);
+        },
+
+        /**
+         * Triggered by the afterLoad event of the lazy loading plugin.
+         *
+         * @param {DOMElement} el
+         * @private
+         */
+        _afterLoad: function (el) {
+            this.trigger('afterLoad', el);
+        },
+
+        /**
+         * Triggered by the onError event of the lazy loading plugin.
+         *
+         * @param {DOMElement} el
+         * @private
+         */
+        _onError: function (el) {
+            this.trigger('onError', el);
+        },
+
+        /**
+         * Triggered by the onFinished event of the lazy loading plugin.
+         *
+         * @param {DOMElement} el
+         * @private
+         */
+        _onFinishedAll: function () {
+            this.all_finished.resolve();
+            this.trigger('onFinishedAll');
```

Comment:
as we said at OCA CS: document how you can hook to JS in README and here just place some tech comments (basically what you've explained me yesterday ;))